### PR TITLE
Remove gogen requirement for feature coverage

### PR DESCRIPTION
**Pull request by therealmitchconnors**

### CODE REVIEW

**Comment by mandarjog** [comment]
probably a bad merge :-( +33k lines....

File: pkg/test/framework/features/features.yaml [hunk 1]
```diff
@@ -16,26 +16,26 @@
 #"values" is a keyword and must contains a list of strings. "values" is omitted when building the label name
 #   This is used when a map needs direct leaf nodes but also other maps
 features:
-  values: 
-    - observability
+  observability:
   traffic:
     circuit-breakers:
-        - fail-gracefully
+        fail-gracefully:
```
Comment (mandarjog) [inline pkg/test/framework/features/features.yaml:22]: is the value a string or `{}` ?

File: pkg/test/framework/test.go [hunk 2]
```diff
@@ -74,8 +78,23 @@ func (t *Test) Label(labels ...label.Instance) *Test {
...
+	log.Error(pwd)
+	c, err := features.BuildChecker(env.IstioSrc + "/pkg/test/framework/features/features.yaml")
+	if err != nil {
+		log.Errorf("Unable to build feature checker: %s", err)
+		t.goTest.FailNow()
+		return nil
+	}
+	for _, f := range feats {
+		if !c.Check(f) {
+			log.Errorf("feature %s is not present in features.yaml", f)
```
Comment (mandarjog) [inline pkg/test/framework/test.go:92]: provide full path in the error.

File: pkg/test/framework/features/features.yaml (hunk 1 above)
Comment (therealmitchconnors) [reply pkg/test/framework/features/features.yaml:22]: The value of a leaf node is nil.  The colon is necessary to be valid yaml, and to make it easy to use an old leaf node as a new branch node.

Comment (mandarjog) [reply pkg/test/framework/features/features.yaml:22]: Thanks please add this in the inline comments, otherwise lgtm

**Pull request merged**